src/components/AllClients/container.js

```diff
@@ -1,49 +1,62 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 import { PropTypes as PT } from 'prop-types';
-import Swal from 'sweetalert2';
-import { getAllClients } from '../../services/clientService';
+import { updateSelectedClientId, fetchAllClients } from '../../store/actions/clients';
+import { getAllClients, getSelectedClientId, getClientsLoadingStatus } from '../../store/reducers';
 
-export default Wrapped =>
+const AllClientsContainer = Wrapped =>
   class extends Component {
     static propTypes = {
-      history: PT.object.isRequired,
+      fetchAllClients: PT.func,
+      isUpdating: PT.bool,
+      selectedClientId: PT.oneOfType([
+        PT.string,
+        PT.number,
+      ]),
+      updateSelectedClientId: PT.func,
+      clients: PT.array,
     };
-    constructor(props) {
-      super(props);
-      this.state = { clients: [], selectedClient: null };
-    }
-
-    componentDidMount() {
-      this.getClients();
-    }
 
-    getClients = () => {
-      getAllClients()
-        .then(response => {
-          const clients = response.data;
-          this.setState({ clients });
-        })
-        .catch(error => {
-          Swal('Error', `Error getting clients: ${error.message}`, 'error');
-        });
-    };
-
-    selectClient = (selectedClient, shouldRefresh) => {
-      this.setState({ selectedClient });
+    selectClientHandler = (updatedClientID, shouldRefresh) => {
+      this.props.updateSelectedClientId(updatedClientID);
       if (shouldRefresh) {
-        this.getClients();
+        this.props.fetchAllClients();
       }
     };
 
     render() {
+
+      const { clients, isUpdating, selectedClientId } = this.props;
+      let selectedClient = clients.filter(({clientId}) => clientId === selectedClientId)[0];
```

Comment:
`Array.find()` would be better to use in this situation where we're after a single element.

```
let selectedClient = clients.find(({clientId}) => clientId === selectedClientId);
```

Reply:
Nice 👍 This has been updated